### PR TITLE
Add run_exports section

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Adding a xeus-exports section. Octave 7 bumbed the soname, which broke downstream packages like xeus-octave.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
@isuruf I picked `max_pin='x.x' by default, but someone more informed about the octave version semantics can chime in and make things looser.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,10 +9,12 @@ source:
...
 
 build:
-  number: 1
+  number: 2
   skip: true  # [win]
-  binary_has_prefix_files:   # [unix]
+  binary_has_prefix_files:   # [linux]
     - bin/octave-{{ version }}  # [linux]
+  run_exports:
+    - {{ pin_subpackage('octave', max_pin='x.x') }}
```
Comment (isuruf) [inline recipe/meta.yaml:17]:
```suggestion
    - {{ pin_subpackage('octave', max_pin='x') }}
```

**Comment by isuruf** [comment]
Please don't merge like that without getting approval from maintainers. You can add yourself as a maintainer if you want to do changes like this in the future.

**Comment by SylvainCorlay** [comment]
> Please don't merge like that without getting approval from maintainers. You can add yourself as a maintainer if you want to do changes like this in the future.

Apologies, I did not realise I wasn't one yet.

**Pull request merged**